### DISCUSSION TITLE
How to filter InterfaceTemplate ObjectVar in custom script?

### DISCUSSION THREAD

**Discussion by llamafilm** [netbox-community/netbox#15535] [Q&A] [answered]
I'm trying to build a custom script to add devices.  First you select a `DeviceType`, and then you select an `InterfaceTemplate` filtered by that.  With the below sample code, I expect the choices in the second var to be filtered after choosing the first.  But the filtering doesn't work; the second var shows all InterfaceTemplates from all device types.

This approach works fine for other object types, for example I can filter `Interfaces` after choosing a `Device`.  `InterfaceTemplate` is the first one I've had a problem with.

```py
from dcim.models import *
from extras.scripts import *

class FormDebug(Script):
    class Meta: 
        """Meta Class."""
        name = "Form Debug"

    device_type = ObjectVar(model=DeviceType)
    
    primary_nic = ObjectVar(
        model=InterfaceTemplate,
        query_params={'device_type_id': '$device_type'}
    )


    def run(self, data, commit):
        self.log_info(f"Entered data: {data}")
```

This filter works fine in nbshell:
```
>>> InterfaceTemplate.objects.filter(device_type_id=118)
<RestrictedQuerySet [<InterfaceTemplate: eth1>, <InterfaceTemplate: eth2>, <InterfaceTemplate: eth3>, <InterfaceTemplate: eth4>]>
```

**Comment by kkthxbye-code** [comment, marked as answer]
The filterset field is called devicetype_id, not device_type_id.

https://github.com/netbox-community/netbox/blob/d2fee886001e4abbcb1b4d9ed3fd32521c820be9/netbox/dcim/filtersets.py#L642C5-L646

**Reply by llamafilm** [reply]
Ah, thank you!  This works.  Do you know if this is documented somewhere, or is it best to always reference the source code when building forms?  Both nbshell and the swagger API docs use `device_type_id`.

**Reply by markkuleinio** [reply]
FWIW, #15410 (Rename certain filters to be consistent)

**Reply by llamafilm** [reply]
That’s odd - that issue says both work, but that’s not what I’m seeing here. Sounds like I will have to change my script after 4.1 is released.

**Reply by markkuleinio** [reply]
They'll both work in 4.0, not in 3.7 yet
